File: ruby/raindrops/raindrops.rb
class Raindrops
  def self.convert(number)
    answer = []
    answer << 'Pling' if (number % 3).zero?
    answer << 'Plang' if (number % 5).zero?
    answer << 'Plong' if (number % 7).zero?
    answer.empty? ? number.to_s : answer.join
  end
end

module BookKeeping
  VERSION = 3
end

File: c/raindrops/src/raindrops.c
#include "raindrops.h"

char convert(char *result, int number)
{
	char *str;
	
	if (number % 3 == 0)
		strcat(result, "Pling");
	if (number % 5 == 0)
		strcat(result, "Plang");
	if (number % 7 == 0)
		strcat(result, "Plong");

	if (strlen(result) == 0)
	{
		asprintf(&str, "%d", number);
		strcat(result, str);
	}
		
	return *result;
}
